### ISSUE TITLE
Credentials plugin configures publications prematurely

### ISSUE THREAD

**Issue by ddimtirov** [etiennestuder/gradle-credentials-plugin#22]
The credentials plugin breaks the ivy-publish and maven-publish plugins by forcing them to finalize their config the moment it is applied. When a user tries to setup publications later, they get unrelated exception that the pubishing extension has already been configured and is immutable at that time.

The fix is simple:

In `CredentialsPlugin.java`, change the `getProjectProperty()` from:

```java
        Map<String, ?> properties = project.getProperties();
        return properties.containsKey(key) ? (String) properties.get(key) : defaultValue;
```

to:

```java
        return project.hasProperty(key) ? (String) project.property(key) : defaultValue;
```
 
You may reproduce the failure with the following build:

```gradle
script {
    repositories.maven { url = repoResolvePlugins }
    dependencies { classpath 'nu.studer:gradle-credentials-plugin:1.0.4' }
}
apply plugin: 'maven-publish'
apply plugin: 'nu.studer.credentials'

publishing {
    publications {
        something(MavenPublication) { artifact file('build.gradle') }
    }
}
```

Here is the gradle error:

```
Cannot configure the 'publishing' extension after it has been accessed.
```

Here is the somewhat truncated stack att he point where the plugin forces init of the `publishing` plugin:

```stacktrace
java.lang.Exception: Stack trace
    at java.lang.Thread.dumpStack(Thread.java:1333)
    at org.gradle.api.internal.plugins.ExtensionsStorage$DeferredConfigurableExtensionHolder.configureNow(ExtensionsStorage.java:227)
    at org.gradle.api.internal.plugins.ExtensionsStorage$DeferredConfigurableExtensionHolder.get(ExtensionsStorage.java:208)
    at org.gradle.api.internal.plugins.ExtensionsStorage.getAsMap(ExtensionsStorage.java:57)
    at org.gradle.api.internal.plugins.DefaultConvention$ExtensionsDynamicObject.getProperties(DefaultConvention.java:286)
    at org.gradle.internal.metaobject.CompositeDynamicObject.getProperties(CompositeDynamicObject.java:79)
    at org.gradle.api.internal.project.DefaultProject$4.create(DefaultProject.java:1039)
    at org.gradle.api.internal.project.DefaultProject$4.create(DefaultProject.java:1037)
    at org.gradle.util.SingleMessageLogger.whileDisabled(SingleMessageLogger.java:200)
    at org.gradle.api.internal.project.DefaultProject.getProperties(DefaultProject.java:1037)
    at nu.studer.gradle.credentials.CredentialsPlugin.getProjectProperty(CredentialsPlugin.java:96)
    at nu.studer.gradle.credentials.CredentialsPlugin.apply(CredentialsPlugin.java:46)
    at nu.studer.gradle.credentials.CredentialsPlugin.apply(CredentialsPlugin.java:27)
```

**Comment by ddimtirov** [comment]
I've raised an https://github.com/gradle/gradle/issues/5682 to fix the `project.getProperties()`, but as this is an easy one and the issue also impacts the older Gradle versions it is worth getting the fix in the plugin too.

**Comment by grv87** [comment]
I had the same problem with other plugins, and adding `enableFeaturePreview('STABLE_PUBLISHING')` into `settings.gradle` fixed it.

**Comment by etiennestuder** [comment]
@ddimtirov Thanks for the great problem report and the solution. Your fix will be in the upcoming version 1.0.5.
